### PR TITLE
"Fix" for dkms

### LINKED ISSUE #17: cannot use dkms to compile kernel modules
Trying to install aur/xone-dkms-git

```
$ sudo dkms build -m xone -v 0.2 -k 5.15.1/armv7l --kernelsourcedir /usr/lib/modules/5.15.1-6/build/

Building module:
cleaning build area...
make -j2 KERNELRELEASE=5.15.1 -C /usr/lib/modules/5.15.1-6/build/ M=/var/lib/dkms/xone/0.2/build...(bad exit status: 2)
Error! Bad return status for module build on kernel: 5.15.1 (armv7l)
Consult /var/lib/dkms/xone/0.2/build/make.log for more information.
```
```
$ cat /var/lib/dkms/xone/0.2/build/make.log
DKMS make.log for xone-0.2 for kernel 5.15.1 (armv7l)
Mon May  2 15:58:21 UTC 2022
make: Entering directory '/usr/lib/modules/5.15.1-6/build'
warning: the compiler differs from the one used to build the kernel
  The kernel was built by: gcc (crosstool-NG 1.23.0.418-d590) 10.2.0
  You are using:           gcc (GCC) 11.2.0
  CC [M]  /var/lib/dkms/xone/0.2/build/transport/wired.o
  CC [M]  /var/lib/dkms/xone/0.2/build/transport/dongle.o
cc1: error: cannot load plugin ./scripts/gcc-plugins/arm_ssp_per_task_plugin.so: ./scripts/gcc-plugins/arm_ssp_per_task_plugin.so: wrong ELF class: ELFCLASS64
make[1]: *** [scripts/Makefile.build:277: /var/lib/dkms/xone/0.2/build/transport/wired.o] Error 1
make[1]: *** Waiting for unfinished jobs....
cc1: error: cannot load plugin ./scripts/gcc-plugins/arm_ssp_per_task_plugin.so: ./scripts/gcc-plugins/arm_ssp_per_task_plugin.so: wrong ELF class: ELFCLASS64
make[1]: *** [scripts/Makefile.build:277: /var/lib/dkms/xone/0.2/build/transport/dongle.o] Error 1
make: *** [Makefile:1868: /var/lib/dkms/xone/0.2/build] Error 2
make: Leaving directory '/usr/lib/modules/5.15.1-6/build'
```

### PR DESCRIPTION
**Pull request by tcamargo**
Fixes #17.

### CODE REVIEW

**Comment by amstan** [comment]
So what was the actual change required? I see you did a bunch of config changes, added "ARCH=arm" (which i'm surprised was required, other arch packages are compiled the same way, on device). Then you built it on an arm system (which I imagined was one of the ways to work that this bug would be solved fully).

What are those .config changes for?

PS: Can you have one commit in the future that does both? And perhaps include the package version in the name. See git history for examples.

File: linux-mister/PKGBUILD [hunk 1]
```diff
@@ -78,6 +78,7 @@ _package() {
   cd "${srcdir}/${_srcname}"
 
   KARCH=arm
+  ARCH=arm
```
Comment (amstan) [inline linux-mister/PKGBUILD:81]: Are you sure this does anything? You only declared a local variable in this function, and you're not using it anywhere. None of the subprocesses (like make) can see it.

**Review by amstan — COMMENTED** [review]
I'm not really sure. Your changes here don't really do much. The big change actually required to fix the bug (well, the most straightforward option) is moving the compilation to a native-compiler, not a cross-compiler. And for that we don't really need anything but the pkrev bump.

On the other hand the compilation time will probably go up. My bar is the 5 min ish time.

I would rather find a way to generate the required plugins even with a cross compiler, so i can keep my fast builds.

File: linux-mister/PKGBUILD (hunk 1 above)
Comment (tcamargo) [reply linux-mister/PKGBUILD:81]: It is not, sorry. The only change should in pkgrel.

File: linux-mister/config [hunk 2]
```diff
@@ -3719,6 +3720,8 @@ CONFIG_INIT_STACK_NONE=y
 # CONFIG_GCC_PLUGIN_STRUCTLEAK_BYREF_ALL is not set
 # CONFIG_INIT_ON_ALLOC_DEFAULT_ON is not set
 # CONFIG_INIT_ON_FREE_DEFAULT_ON is not set
+CONFIG_CC_HAS_ZERO_CALL_USED_REGS=y
```
[the remark this answers is not in the document]
Comment (tcamargo) [reply linux-mister/config:3723]: But then I cannot do unattended builds. Suggestion?

**Comment by tcamargo** [comment]
> I'm not really sure. Your changes here don't really do much. The big change actually required to fix the bug (well, the most straightforward option) is moving the compilation to a native-compiler, not a cross-compiler. And for that we don't really need anything but the pkrev bump.

pkgrel and config changes. Disregards the ARCH thing.

> On the other hand the compilation time will probably go up. My bar is the 5 min ish time.

Could you try to generate mister-headers inside chroot? I only have 2 cores assigned to my archlinux vm.

> I would rather find a way to generate the required plugins even with a cross compiler, so i can keep my fast builds.

What would be the goal to have mister-headers with x86_64 plugins? I think any dkms compilation will happen inside chroot.

**Pull request closed without merging**